### ISSUE TITLE
Overloads on `__init__` raises

### ISSUE THREAD

**Issue by haakonvt** [jsh9/pydoclint#255]
We have some trouble with overloads on `__init__` methods. We get `DOC105: ...` for what we think should be supported/correct code:

```
DOC105: Method `FooWrite.__init__`: Argument names match, but type hints in these args do not match: bar, baz
```

Here is the offending code:

```py
class FooWrite(SomeObject):
    """
    The Foo write class.

    Args:
        bar (str | None): Description of bar.
        baz (int | None): Description of baz.
    """

    @overload
    def __init__(self, bar: None, baz: None) -> NoReturn: ...

    @overload
    def __init__(self, bar: str | None, baz: int | None) -> None: ...

    def __init__(self, bar: str | None, baz: int | None) -> None:
        if bar is None and baz is None:
            raise ValueError
```

If we instead do the exact same on a different method, everything works as expected:

```py
class FooWrite(SomeObject):
    @overload
    def method(self, bar: None, baz: None) -> NoReturn: ...

    @overload
    def method(self, bar: str | None, baz: int | None) -> None: ...

    def method(self, bar: str | None, baz: int | None) -> None:
        """
        The method method

        Args:
            bar (str | None): Description of bar.
            baz (int | None): Description of baz.
        """
        if bar is None and baz is None:
            raise ValueError
```

Any idea? I suspect there's something going wrong since we look for/demand the docstring be place on the class instead of the method.

**Some more info**: We run pydoclint as part of pre-commit, here is the config we use:

```
  - repo: https://github.com/jsh9/pydoclint
    rev: 0.6.7
    hooks:
      - id: pydoclint
        require_serial: true
        args:
          - --style=google
          - --exclude=tests/|scripts/
          - --quiet
          - --skip-checking-raises=true
          - --check-class-attributes=false
```

**Comment by jsh9** [comment]
Hi @haakonvt ,

This seems like an expected behavior (I'm not saying it's the best behavior though).

What's happening under the hood (for your class constructor example) is: pydoclint is checking all 3 class constructors against the class docstring, and it's expecting all 3 constructors to have arg type hints consistent with the docstring.

That explains why your 2nd example passes without violations.

The solution I can think of is to write separate docstrings for those class constructors, and set `--allow-init-docstring` to True:

```python

class FooWrite(SomeObject):
    """
    The Foo write class.
    """

    @overload
    def __init__(self, bar: str, baz: str) -> NoReturn:
        """
        Something

        Args:
            bar (str): Description of bar.
            baz (str): Description of baz.
        """
        ...

    @overload
    def __init__(self, bar: list, baz: list) -> None:
        """
        Something

        Args:
            bar (list): Description of bar.
            baz (list): Description of baz.
        """
        ...

    def __init__(self, bar: dict, baz: dict) -> None:
        """
        Something

        Args:
            bar (dict): Description of bar.
            baz (dict): Description of baz.
        """
        if bar is None and baz is None:
            raise ValueError

```

**Comment by haakonvt** [comment]
I disagree - there is one class init/constructor method, and then several `typing.overload`s are necessary to indicate that passing both parameters as `None` will raise (i.e. `typing.NoReturn`).

If it works for a generic method it should work for `__init__` (...or both should not work).

**Comment by jsh9** [comment]
Hi @haakonvt , I've fixed this issue. It's published as 0.8.2.

**Comment by haakonvt** [comment]
@jsh9 amazing! 🚀

**Comment by baluyotraf** [comment]
Just want to say thanks for the fix 👍
